### PR TITLE
Mason: Compile options in toml file

### PR DESCRIPTION
**Pull request by Spartee**
This PR is a feature for mason that allows users to put their compopts right in the ``Mason.toml`` file as listed in issue #7106. When this feature is used, the format for the ``Mason.toml`` will be a follows:

``` toml
[brick]
name = "MyProject"
version = "0.1.0"
compopts = "-M /test/somePackage"

[dependencies]
```

This is a step towards being able to support C dependencies. I do not believe this will conflict with the changes that @daviditen is proposing in #7310 but I am mentioning him in this just in case. 

This change has passed testing on my linux64 machine.

### CODE REVIEW

File: tools/mason/MasonUtils.chpl [hunk 1]
```diff
@@ -31,6 +33,16 @@ proc getEnv(name: string): string {
 }
 
 
+/* Retrieves the compilation flags from lock file */
+proc getCompopts(lock: Toml) {
+  if lock.pathExists('root.compopts') {
+    var compopts = lock['root']['compopts'].s;
```
Comment (benharsh) [inline tools/mason/MasonUtils.chpl]: I don't think you need the temporary 'compopts' here.

**Comment by benharsh** [comment]
Does the 'compopts' field automatically get copied to 'root'?

File: tools/mason/MasonUtils.chpl [hunk 2]
```diff
@@ -31,6 +33,16 @@ proc getEnv(name: string): string {
...
+    return compopts;
+  }
+  else return ' ';
```
Comment (daviditen) [inline tools/mason/MasonUtils.chpl]: Why not return an empty string instead of a string containing a space?

**Comment by daviditen** [comment]
I don't think this and #7310 conflict, so you should be good on that front.

**Comment by Spartee** [comment]
@benharsh requesting a second review whenever you have time

**Pull request merged**